File: app/src/main/java/br/com/star_wars_wiki/view_model/FavoriteViewModel.java
package br.com.star_wars_wiki.view_model;

import android.app.Application;
import android.content.Context;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

import br.com.star_wars_wiki.database.repository.FavoriteRepo;
import br.com.star_wars_wiki.entity.Favorite;

public class FavoriteViewModel extends AndroidViewModel {
    private FavoriteRepo favoriteRepo;
    private LiveData<List<Favorite>> getAllFavorite;

    public FavoriteViewModel(@NonNull Application application) {
        super(application);
        favoriteRepo = new FavoriteRepo(application);
        getAllFavorite = favoriteRepo.getAllFavorites();
    }

    public void insert(Favorite favorite){
        favoriteRepo.insert(favorite);
    }

    public void remove(Favorite favorite, Context context){
        favoriteRepo.remove(favorite, context);
    }

    public LiveData<List<Favorite>> getAllFavorites(){
        return this.getAllFavorite;
    }
}

File: app/src/main/java/com/matheussilas97/starwarsapp/view/favorites/FavoriteViewModel.kt
package com.matheussilas97.starwarsapp.view.favorites

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.matheussilas97.starwarsapp.api.ApiFactoryFavorites
import com.matheussilas97.starwarsapp.api.service.FavoritesService
import com.matheussilas97.starwarsapp.database.model.FavoriteModel
import com.matheussilas97.starwarsapp.database.repository.FavoriteRepository

class FavoriteViewModel (application: Application) : AndroidViewModel(application) {

   private val service = ApiFactoryFavorites.create(FavoritesService::class.java)
    private val favoriteRepository = FavoriteRepository(application.applicationContext)

    private val mFavoriteList = MutableLiveData<List<FavoriteModel>>()
    val favoriteList: LiveData<List<FavoriteModel>> = mFavoriteList

    fun list() {
        mFavoriteList.value = favoriteRepository.listFavorites()
    }

    fun deleteFavorite(id: String) {
        val favorite = favoriteRepository.getLoad(id)
        if (favorite != null) {
            favoriteRepository.deleteStudents(favorite)
        }
    }


}